modules/nomad-cloudinit-ubuntu-v1/templates/nomad_user_data.tpl

```diff
@@ -39,9 +39,11 @@ echo "--------------------------------------"
 echo "      Creating config.hcl"
 echo "--------------------------------------"
 export PRIVATE_IP="$(/sbin/ifconfig eth0 | grep 'inet addr:' | cut -d: -f2 | awk '{ print $1}')"
+export INSTANCE_ID="$(curl http://169.254.169.254/latest/meta-data/instance-id)"
```

Comment:
That ip address seemed to be a mistake? @teresaibarra

Reply:
@junejung I should've added more comments here -- my mistake
This `curl` command retrieves the instance ID from this URL, as noted in the AWS documentation [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html). This instance's information is available from within the running instance, so we just grab the information from that IP address.

Hashicorp does similarly in this [file](https://github.com/hashicorp/terraform-aws-nomad/blob/master/modules/run-nomad/run-nomad#L9).

It is kinda weird -- though it is nice that we aren't billed for it. Should I add a comment here about how we get the instance ID?

Reply:
Pull that hard-coded IP address out and replace with a variable in the top of the file?